### PR TITLE
[mongodb] Added ChangeStream class

### PR DESCRIPTION
**Pull request by mantyz**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<http://mongodb.github.io/node-mongodb-native/3.0/api/ChangeStream.html>>

### CODE REVIEW

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -1415,3 +1417,40 @@ export interface GridFSBucketWriteStreamOptions {
...
     j?: number
 }
+
+/** http://mongodb.github.io/node-mongodb-native/3.0/api/ChangeStream.html */
+export class ChangeStream extends Readable {
+    constructor(changeDomain: Db | Collection, pipeline: Object[], options?: ChangeStreamOptions);
+
+    /** http://mongodb.github.io/node-mongodb-native/3.0/api/ChangeStream.html#close */
+    close(): Promise<any>;
+    close(callback: ChangeStreamCallback<any>): void;
```
Comment (CaselIT) [inline types/mongodb/index.d.ts]: Can use the `MongoCallback` interface instead of the `ChangeStreamCallback` so we avoid a new interface?

**Review by CaselIT — CHANGES_REQUESTED** [review]

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (mantyz) [reply types/mongodb/index.d.ts]: I agree. Will be fixed.

**Review by CaselIT — APPROVED** [review]
LGTM now

**Review by weswigham — CHANGES_REQUESTED** [review]
@mantyz This PR has some merge conflicts; could you resolve them? Thanks.

**Comment by mantyz** [comment]
@weswigham Conflicts fixed.

**Pull request merged**